src/core/mesh/qgsmeshvirtualdatasetgroup.cpp

```diff
@@ -152,7 +154,12 @@ bool QgsMeshVirtualDatasetGroup::calculateDataset() const
   if ( !mLayer )
     return false;
 
-  const QgsMeshCalcUtils dsu( mLayer, mDatasetGroupNameUsed, QgsInterval( mDatasetTimes[mCurrentDatasetIndex] / 1000.0 ) );
+  const QgsMeshCalcUtils dsu( mLayer,
```

Comment:
I do not understand the concept before this bug fix. Is the problem that the virtual dataset group has ONLY one timestep calculated (first?) and therefore the calculator does not do aggregation? 

Isn't the actual problem here that the virtual dataset group is lazy loading datasets, and in the context of the mesh calculator it should just state how many timesteps it has in total and not as already loaded? 
Or when doing a deep copy of the virtual dataset, we can calculate ALL timesteps in the mesh calculator?

Reply:
For non-virtual datasets it works "before" this PR? If yes, how it works there and what is the difference between regular group and virtual?